auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -359,7 +363,9 @@ public Void then(@NonNull Task<GoogleApiClient> task) throws Exception {
      * silently.
      *
      * @param activity the calling {@link Activity}.
+     * @deprecated use {@link #delete(FragmentActivity)} instead
      */
+    @Deprecated
     public Task<Void> delete(@NonNull Activity activity) {
```

Comment:
As far as I can tell the difference between `delete(FragmentActivity)` and `delete(Activity)` is just that the former uses `CredentialApiHelper` and the latter uses `OldCredentialApiHelper`.

Could this all be simplified by having both of the `CredentialApiHelper` classes implement a common interface?  I think it would look like this:

```
interface CredentialApiHelper {

  Task<Status> delete(Credential credential);

  Task<Status> disableAuthSignIn();

}
```

And then these two `delete()` functions would just be a matter of constructing the proper `Helper` and then calling to a private `Task<Void> delete(CredentialApiHelper helper)`